### PR TITLE
Only create and upgrade if user does not have preexisting MHV account.

### PR DESCRIPTION
**Pull request by U-DON**
Resolves department-of-veterans-affairs/vets.gov-team#5892.

### CODE REVIEW

File: app/models/mhv_account.rb [hunk 1]
```diff
@@ -69,7 +73,7 @@ def terms_and_conditions_accepted?
   end
 
   def preexisting_account?
-    user&.mhv_correlation_id.present?
+    user&.mhv_correlation_id.present? && !previously_registered?
```
Comment (U-DON) [inline app/models/mhv_account.rb]: I've redefined `preexisting_account?` to mean that the user has an `mhv_correlation_id` not from registering through Vets.gov. I noticed that this method hasn't been used anywhere, so I think the impact should be isolated to these changes.

File: spec/models/mhv_account_spec.rb [hunk 2]
```diff
@@ -247,11 +247,12 @@
 
     context 'existing account that has not been upgraded' do
       let(:mhv_ids) { ['14221465'] }
-      let(:base_attributes) { { user_uuid: user.uuid, account_state: 'unknown' } }
+      let(:base_attributes) { { user_uuid: user.uuid, account_state: 'registered' } }
 
       it 'will only upgrade an account and set the time the account was upgraded' do
+        subject = described_class.new(base_attributes.merge(registered_at: Time.current))
```
Comment (U-DON) [inline spec/models/mhv_account_spec.rb]: How come I have to declare `registered_at` with this merge? I tried adding it to the hash above at first, but it didn't work until I used `merge`.

File: spec/models/mhv_account_spec.rb [hunk 3]
```diff
@@ -281,43 +281,20 @@
...
           expect(subject.account_state).to eq('upgraded')
-          expect(subject.registered_at).to be_nil
           expect(subject.upgraded_at).to be_a(Time)
           expect(subject.eligible?).to be_truthy
           expect(subject.terms_and_conditions_accepted?).to be_truthy
         end
       end
     end
 
-    context 'existing account that has already been upgraded' do
```
Comment (U-DON) [inline spec/models/mhv_account_spec.rb]: This scenario shouldn't have been possible anyway, even before my changes. If an account is in the `upgraded` state, `may_upgrade` should return `false` for the conditional block where `mhv_ac_client.post_upgrade` occurs, so it shouldn't be possible to reach the `rescue` block.

File: app/models/mhv_account.rb [hunk 4]
```diff
@@ -56,8 +67,11 @@ class MhvAccount < ActiveRecord::Base
   end
 
   def create_and_upgrade!
-    create_mhv_account! unless preexisting_account?
-    upgrade_mhv_account!
+    # Accept T&C if existing?
```
Comment (U-DON) [inline app/models/mhv_account.rb]: Should we persist any T&C acceptance info if the account is preexisting? Probably not?

File: spec/models/mhv_account_spec.rb (hunk 2 above)
Comment (saneshark) [reply spec/models/mhv_account_spec.rb]: When vets.gov creates an account, we also set the time the account is registered_at. Similarly, we set upgraded_at when the account gets upgraded. I don't think we have validations on the model though, so it must be the spec that is testing for this.

File: spec/models/mhv_account_spec.rb (hunk 3 above)
Comment (saneshark) [reply spec/models/mhv_account_spec.rb]: The problem is that the account could be "upgraded" / "premium" and we don't know that it is or not until we make the call to upgrade. This test was there to satisfy that particular scenario.

File: app/models/mhv_account.rb [hunk 5]
```diff
@@ -56,8 +67,11 @@ class MhvAccount < ActiveRecord::Base
...
+    unless existing?
+      create_mhv_account! unless previously_registered?
```
Comment (saneshark) [inline app/models/mhv_account.rb]: existing means the account has an mhv_correlation_id and previously_registered? means that vets.gov registered it? I would like for these method names to be a little more clear to that point maybe. It's a little confusing to think about it, but I think I get the idea.

File: app/models/mhv_account.rb (hunk 4 above)
Comment (saneshark) [reply app/models/mhv_account.rb]: I think we should be OK on that. Technically the T&C will hopefully get moved to the account creation controller anyway and would no longer need to be tied to an mhv_account state at all in the future.

File: spec/models/mhv_account_spec.rb (hunk 3 above)
Comment (U-DON) [reply spec/models/mhv_account_spec.rb]: So should I re-introduce this case?

Comment (saneshark) [reply spec/models/mhv_account_spec.rb]: I think we should include this case, yes. Perhaps update the description to reflect what I just described if its not clear.

Comment (U-DON) [reply spec/models/mhv_account_spec.rb]: Don't we know whether it's upgraded based on `previously_upgraded?` (or `upgraded_at`)?

Comment (U-DON) [reply spec/models/mhv_account_spec.rb]: Wouldn't we know what it is based on `upgraded_at` (and thus `previously_upgraded`)?

**Comment by U-DON** [comment]
Since this PR seems to be in a satisfactory place, I'm going to merge it to the `remove_upgrade` branch. That's the branch that will be merged into master when it's ready for staging.

**Comment by saneshark** [comment]
OK, do you mind closing and opening that branch against master when you have a moment? This one here shows file diff that's a little inconsistent with whats actually modified.

**Pull request merged**